### PR TITLE
Add Agent config file mounting and agent cli arg parameter.

### LINKED ISSUE #10: Add agent config volume
A volume needs to exist to hold the agent config file.  This volume should be a config map volume.

The agent should have a volume mount that binds the agent to the volume. The agent command line parameters should reference the agent-configmap's agent configuration as a file at its mounted location.

It is unclear if there is a need to specify the volume mount path, but for biodegradability, the volume name and mount path names should not be configurable.

### PR DESCRIPTION
**Pull request by edwbuck**
Closes #10

agent-configmap now has the data property for the configuration informaiton set dynamically by the file at the end of agent.configFile.

Test updates to set the basic default of the config file name to agent.conf which requires a different style of json path specification to accomodate the internal period in the path name.

### CODE REVIEW

File: CONFIGURATION.md [hunk 1]
```diff
@@ -6,6 +6,21 @@
...
+## File locations
+
+Some file in the spire deployment can be relocated.  Relocating files may assist
+in avoiding collisions when installing multiple SPIRE instances on the same
+hardware, or may place files into locations confroming with your file placement
+standards.  Below is a list of relocatable files and their default locations.
+
+| File                          | Config Path      | Default Location           | 
+| ----------------------------- | ---------------- | -------------------------- |
+| Agent Configuration File      | agent.configFile | /opt/spire/conf/agent.conf |
```
[the remark this answers is not in the document]
Comment (edwbuck) [reply CONFIGURATION.md:19]: I am fond of LSB.  In this case I thought "something like the SPIRE documentation" was a goal with higher priority than LSB.

If you feel differently, I'll happily change the default, or confirm that this is ok.

File: spire-flex/templates/agent-daemonset.yaml [hunk 2]
```diff
@@ -31,4 +39,8 @@ spec:
               host: {{ default "localhost" (((.Values.agent).healthCheck).bindAddress) | quote }}
               port: {{ default 80 (((.Values.agent).healthCheck).bindPort) }}
               path: {{ default "/ready" (((.Values.agent).healthCheck).readyPath) | quote }}
-         {{- end }}
+          {{- end }}
+      volumes:
+        - name: {{ $.Release.Name }}-agent-config-volume
+          configMap:
+            name: {{ $.Release.Name }}-agent-config
```
[the remark this answers is not in the document]
Comment (edwbuck) [reply spire-flex/templates/agent-daemonset.yaml:46]: Just a holdover of not wanting five "foo agent" names that then one needs to decide if it's a config map, daemonset, POD, service account, or something else.

I reserve "foo agent" for the POD.  If you feel differently, please share a proposed naming pattern.

File: spire-flex/templates/agent-daemonset.yaml [hunk 3]
```diff
@@ -15,6 +15,14 @@ spec:
       containers:
         - name: {{ $.Release.Name }}-agent
           image: {{ include "spire-flex.agent.image" . }}
+          args:
+            - "-config"
+            - {{ default "/opt/spire/conf/agent/agent.conf" ($.Values.agent).configFile }}
+          volumeMounts:
+            - name: {{ $.Release.Name }}-agent-config-volume
+              mountPath: {{ default "/opt/spire/conf/agent/agent.conf" ($.Values.agent).configFile }}
```
[the remark this answers is not in the document]
Comment (edwbuck) [reply spire-flex/templates/agent-daemonset.yaml:23]: You are right, the default needs to omit the trailing "agent.conf" and the ($.Values.agent).configFile needs to be trimmed to its directory path.

Expect an update today.

File: CONFIGURATION.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kfox1111) [reply CONFIGURATION.md:19]: 🤷

File: spire-flex/templates/agent-daemonset.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (edwbuck) [reply spire-flex/templates/agent-daemonset.yaml:46]: I get that the wrong kinds of redundancy are clunky and unwanted.  I'm ok with making the non-referenced names a bit more "agent-config" and a bit less "agent-config-volume / agent-config-pvc / etc."  

The "foo-config" comes with a flaw in the argument, basically by making something abstract, it's hard to rationalize about how the concrete pattern should apply.  It is heavily dependent on what the abstract thing is in the non-refrenced concrete example.

If we say "agent-config" I think as long as all the items involved are somehow related to the Agent Config File, I am ok.  If we need to pivot when we support Agent Config PVCs, then we can pivot then.

I'm making the update to the templates.  Expect a push in a minute or two.

File: spire-flex/templates/agent-daemonset.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (edwbuck) [reply spire-flex/templates/agent-daemonset.yaml]: updated, this comment is now out of date

File: spire-flex/templates/agent-daemonset.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kfox1111) [reply spire-flex/templates/agent-daemonset.yaml:46]: That misses the point. I strongly dislike, as a user, having to kubectl get configmap foo-config. Whats the volume is internally called is irrelevant. That's not a compromise  I don't as a user ever have to type it. Feel free to disregard it, but I'm sure I am not the only person to have a strong dislike for this kind of naming of objects. This is just a straw though.

We're getting to be in a bad place here.... I keep giving my opinions on things, and it feels like the majority of things are just blown off, decided without me, or your own pr proposed to do it your own way instead. That's fine. you can do what you want. but again, I have little incentive to contribute if it keeps happening. We need to compromise and work together, or we can go our separate ways.

**Review by kfox1111 — APPROVED** [review]

**Pull request merged**